### ISSUE TITLE
terraform-provider-azuredevops_v1.3.0 plugin crashed while deploying 'azuredevops_agent_queue'

### ISSUE THREAD

**Issue by jrai-199** [microsoft/terraform-provider-azuredevops#1183]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version
v.1.9.4

### Affected Resource(s)

* `azuredevops_agent_queue` , `azuredevops `

### Terraform Configuration Files

```hcl
This is the module call
module "managed_devops_pool" {
  source                         = "../core_module"
  resource_group_name            = azurerm_resource_group.this.name
  location                       = azurerm_resource_group.this.location
  name                           = var.managed_devops_pool_name
  dev_center_project_resource_id = azurerm_dev_center_project.this.id
  subnet_id                      = data.azurerm_subnet.mdp_subnet.id
  organization_profile = {
    organizations = [{
      name     = var.azure_devops_organization_name
      projects = [data.azuredevops_project.this.name]
    }]
  }
  #enable_telemetry = var.enable_telemetry
  /* diagnostic_settings = {
    sendToLogAnalytics = {
      name                           = "sendToLogAnalytics"
      workspace_resource_id          = azurerm_log_analytics_workspace.this.id
      log_analytics_destination_type = "Dedicated"
    }
  } */
  tags = local.tags
  depends_on = [
    azapi_resource_action.resource_provider_registration,
    #data.virtual_network
  ]
}



# #Use this for existing Devops Project #
data "azuredevops_project" "this" {
  name = var.azuredevops_project_name
}
**resource "azuredevops_agent_queue" "this" {
  project_id      = data.azuredevops_project.this.id
  name            = module.managed_devops_pool.name
  depends_on      = [module.managed_devops_pool]
}**

resource "azuredevops_pipeline_authorization" "this" {
  project_id  = data.azuredevops_project.this.id
  resource_id = azuredevops_agent_queue.this.id
  type        = "queue"
}

resource "azurerm_resource_group" "this" {
  location = local.selected_region
  name     = var.azurerm_resource_group_name
}

output "managed_devops_pool_id" {
  value = module.managed_devops_pool.resource_id
}

output "managed_devops_pool_name" {
  value = module.managed_devops_pool.name
}
```

### Debug Output
https://gist.github.com/jrai-199/92c53bc6caaff5bf5076a866a4e5c8e5

### Expected Behavior
Managed devops pool should be added to Devops Project.

### Actual Behavior
During terraform apply, terraform plugin crashed  due to agent queue resource .
NO managed devops pool was added to devops project. 

### Steps to Reproduce
1. Terraform plan
2. Terraform apply

### References

* #0000

**Comment by brandonavant** [comment]
I am running into this same issue. If I attempt to perform a `terraform plan` for IaC that was working a week or so ago, I get a very similar crash.

**Comment by brandonavant** [comment]
@jrai-199 , while I definitely agree that there's a problem here that needs to be fixed in the provider, I was able to get around this issue by fixing issues in my state file.

I had semi corrupted my state by cycling between manual changes (adds, edits, and deletes) in both ADO and Azure and running my Terraform. Somehow my state was referencing things that no longer existed. For some reason, this was causing the crashes, whenever it tried to compare the state to the actual resources in ADO and Azure.

I'm not sure if this is the scenario you're in but understanding how to read and interpret the state file might be helpful. 

For this, please research the following commands:

- `terraform state list`: List all resources in the state file
- `terraform state show <resource-id>`: Show details of a resource in the state file
- `terraform state rm <resource-id>`: Remove a resource from the state file
- `terraform import <resource-id> <resource-name>`: Import a resource into the state file

Again, this might not be relevant for your scenario and there's still some error handling missing from the provider but perhaps this _might_ be helpful for you as it was for me.

**Comment by xuzhang3** [comment]
The error seems to be the `var.managed_devops_pool_name` is nil, can you help check the local configurations? The ADO provider should detect this and will enforce validation.

**Comment by SOLDIERz** [comment]
Facing the same Issue. Seems that Microsoft changed something in their API. Yesterday everything was fine. Facing this Issue with "azuredevops_branch_policy_min_reviewers" Resource.

```
Stack trace from the terraform-provider-azuredevops_v1.3.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc6f178]

goroutine 371 [running]:
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils.(*SecurityNamespace).GetPrincipalPermissions.func1({0xf97300?, 0xc0006b3e60?})
	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils/namespaces.go:442 +0x98
reflect.Value.call({0xe19960?, 0x102b3b8?, 0x18?}, {0xfbfa0e, 0x4}, {0xc000141230, 0x1, 0xc000141230?})
	reflect/value.go:596 +0xca6
reflect.Value.Call({0xe19960?, 0x102b3b8?, 0xc0006b3dd0?}, {0xc000141230?, 0x7f0f6ca245b8?, 0x18?})
	reflect/value.go:380 +0xb9
github.com/ahmetb/go-linq.(*genericFunc).Call(0xc00012ff58, {0xc0007eed10, 0x1, 0x1?})
	github.com/ahmetb/go-linq@v3.0.0+incompatible/genericfunc.go:35 +0x10b
github.com/ahmetb/go-linq.Query.SelectT.func1({0xf97300?, 0xc0006b3e60?})
	github.com/ahmetb/go-linq@v3.0.0+incompatible/select.go:47 +0x45
github.com/ahmetb/go-linq.Query.SelectT.Query.Select.func3.1()
	github.com/ahmetb/go-linq@v3.0.0+incompatible/select.go:24 +0x3b
github.com/ahmetb/go-linq.Query.ToSlice({0xded340?}, {0xdea240?, 0xc0001411b8?})
	github.com/ahmetb/go-linq@v3.0.0+incompatible/result.go:634 +0x1bc
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils.(*SecurityNamespace).GetPrincipalPermissions(0xc0007af620, 0xc0007ef180)
	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils/namespaces.go:444 +0xe5
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils.GetPrincipalPermissions(0xc00052b600, 0xc0007af620)
	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils/securityHelper.go:106 +0x105
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions.resourceProjectPermissionsRead(0xc00052b600, {0xdc2840?, 0xc00042aea0?})
	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/resource_project_permissions.go:61 +0x72
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1136328?, {0x1136328?, 0xc0004ca900?}, 0xd?, {0xdc2840?, 0xc00042aea0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:712 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0004af420, {0x1136328, 0xc0004ca900}, 0xc000629040, {0xdc2840, 0xc00042aea0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:1015 +0x51d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0003f3638, {0x1136328?, 0xc0004ca660?}, 0xc0000ad380)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/grpc_provider.go:613 +0x4aa
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0002c12c0, {0x1136328?, 0xc00057dd40?}, 0xc0007af380)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:748 +0x488
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xf78fc0, 0xc0002c12c0}, {0x1136328, 0xc00057dd40}, 0xc00022c070, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000168000, {0x113abc0, 0xc0004b2680}, 0xc000698480, 0xc000428f00, 0x1772af0, 0x0)
	google.golang.org/grpc@v1.56.3/server.go:1335 +0xdb8
google.golang.org/grpc.(*Server).handleStream(0xc000168000, {0x113abc0, 0xc0004b2680}, 0xc000698480, 0x0)
	google.golang.org/grpc@v1.56.3/server.go:1712 +0x9da
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.3/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 14
	google.golang.org/grpc@v1.56.3/server.go:958 +0x136

Error: The terraform-provider-azuredevops_v1.3.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Also face the same Issue with AzureDevOps Provider in version 1.1.1. When I'm executing on an Repository with no Changes everything works fine.**

**Comment by xuzhang3** [comment]
The `azuredevops_agent_queue` issue has been fixed in PR #1157  ,  will be available in `v1.4.0`

**Comment by SOLDIERz** [comment]
@xuzhang3 when will 1.4.0 be released?

**Comment by xuzhang3** [comment]
@SOLDIERz From the error log, this issue is related to `resource_project_permissions`, can you open another issue for this issue?

**Comment by xuzhang3** [comment]
> [@xuzhang3](https://github.com/xuzhang3) when will 1.4.0 be released?

It will be released at the end of October.

**Comment by jrai-199** [comment]
> The error seems to be the `var.managed_devops_pool_name` is nil, can you help check the local configurations? The ADO provider should detect this and will enforce validation.



This variable is defined in terraform.tfvars and so it is not null.  
From TF plan, variables are instantisated with values as mentioned below :

# azuredevops_agent_queue.this will be created
  + resource "azuredevops_agent_queue" "this" {
      + agent_pool_id = (known after apply)
      + id            = (known after apply)
      + name          = "ag-******-01"
      + project_id    = "******68cc***18"
    }

**Comment by xuzhang3** [comment]
@jrai-199 this is a know issue and has been fixed in #1157
